test/integration/main_test.rb

```diff
@@ -10,7 +10,13 @@ class MainTest < IntegrationTest
 
     assert_app_is_down
 
-    kamal :deploy
+    mock = Minitest::Mock.new
```

Comment:
I don't think we should be mocking things in the integration tests. We should just test the results of the commands.

In any case `kamal :deploy` actually runs `docker compose exec deployer kamal deploy`, so I'm not sure the mocking is doing anything.

We could add a second redeploy of the second version later on and add new `assert_hooks_did_not_run` assertion after it to confirm the pre-build hook was not called, which would be enough to infer that there was no rebuild.